torch/utils/checkpoint.py

```diff
@@ -194,25 +194,16 @@ def set_device_states(devices, states) -> None:
 
 
 def _get_autocast_kwargs(device="cuda"):
-    if device == "cuda":
-        device_autocast_kwargs = {
-            "enabled": torch.is_autocast_enabled(),
-            "dtype": torch.get_autocast_gpu_dtype(),
-            "cache_enabled": torch.is_autocast_cache_enabled(),
-        }
-    elif _supports_autocast(device):
-        device_module = _get_device_module(device)
-        device_autocast_kwargs = {
-            "enabled": device_module.is_autocast_enabled(),
-            "dtype": device_module.get_autocast_dtype(),
-            "cache_enabled": torch.is_autocast_cache_enabled(),
-        }
-    else:
-        device_autocast_kwargs = None
+    assert device != "cpu"
+    device_autocast_kwargs = {
+        "enabled": torch.is_autocast_enabled(device),
```

Comment:
The deprecated APIs will raise a warning as an error in some dynamo-related UTs. So use device-agnostic APIs instead.